ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx

```diff
@@ -192,6 +128,7 @@ const CgnDiscountDetailHeader = ({ discount }: Props) => (
 
 export const useCgnDiscountDetailBottomSheet = (
   discount: Discount,
+  merchantType?: DiscountCodeType,
   landingPageHandler?: (url: string, referer: string) => void
 ) => {
```

Comment:
This hook looks like a nice candidate (self-contained, yet not trivial) for at least a unit test going through the happy path.
I see some scenarios that are worth testing:
1. ensure that `discount` is propagated and its nullable props are honoured (see `condition` or `landingPageUrl` for instance)
2. ensure that `merchantType` is propagated and handled correctly

On a side note, this being a _hook_ I'd suggest you move it to a `ts/features/bonus/cgn/hook` folder like we do in the app.